include/maidsafe/crux/socket.hpp

```diff
@@ -253,34 +254,32 @@ inline void socket::close() {
     keepalive_timer.stop();
     transmit_queue.shutdown();
 
+    auto m = multiplexer;
```

Comment:
@Fraser999, the multiplexer won't be destroyed because the pin is a shared_pointer<multiplexer> so it will preserve its lifetime until at least all the copies of the pin are destroyed.